### DISCUSSION TITLE
@zos/devices GetDeviceInfo, emulator, and support for "Bip S" model?

### DISCUSSION THREAD

**Discussion by tozachroberts** [zepp-health/zeppos-forum#348] [Developer Feedback]
I am slowly getting comfortable with the rapid app tools like autogui and easystorage, which have been great so far. But I find myself kind of limited by a small mismatch in what is supported in the library and the emulator and the device I actually own. How would I go about contributing to or requesting support for what was sold in the US as the "Amazfit Bip S" model, which feels closest in specification to the "Bip 5 S" but isn't exactly the same? Specifically, I'd like emulator support, and support in the @zos/devices library. I am willing to contribute patches or MRs but I don't think this lib is open source or I am missing its repo location.

**Comment by silver-zepp** [comment]
Unfortunately, `Bip S` doesn't have ZeppOS on board. You can see all the devices that are capable of running it [here](https://docs.zepp.com/docs/reference/related-resources/device-list).

**Reply by tozachroberts** [reply]
Definitely unfortunate. I love the screen on the Bip S and was hoping to extend its value with a few light apps I'd like to have. I was optimistic that since it wasn't listed in the Non-Zepp OS device list (https://docs.zepp.com/docs/reference/related-resources/device-list#non-zepp-os-devices) either, and it was similar in spec to the Bip 5, that maybe that was a model variation only and that I'd find a way to get apps compiled and deployed to it. Thanks as always for the prompt info.

**Reply by silver-zepp** [reply]
No worries. You can see a comparison between those two, here: [Bip S](https://www.notebookcheck.net/The-Amazfit-Bip-S-will-launch-on-June-3.466295.0.html) vs [Bip 5](https://www.notebookcheck.net/Amazfit-Bip-5-New-smartwatch-launches-for-under-US-100-with-1-91-inch-display-and-10-days-of-battery-life.738089.0.html).

It's not on the list as it's a pretty dated device that was launched way before `zos`. It's a decent device, though. 
I love the transflective display, but it's way behind spec-wise to run an operating system.

**Reply by tozachroberts** [reply]
I figured I'd run into some issues, but the more I learned about the structure of zos apps and implementation, the more hope I gave myself. The menu, and the structure of apps with pages and widgets feels very similar to the UI of whatever "operating system" is running on the Bip S. The loose compatibility of many community watch faces added to my hopes. But all good things must end, I'll likely give the Bip 5 Unity a try and enjoy its more open ecosystem and the tremendous library of dev tools you (and a team?) have provided and support. Of course, if there are useless assets that Zepp might choose to open source, like the antiquated firmware and native app / OS of the Bip S, I'd welcome keeping that device useful just because it is such a rare gem. Thanks again @silver-zepp.

**Reply by silver-zepp** [reply]
Thanks for the feedback @tozachroberts. Bip 5 Unity is a perfect pick now that it has the most recent OS.
